transformations/replace_financial_amounts/transformation.py

```diff
@@ -0,0 +1,58 @@
+from typing import List
+import random
+
+from transformations.replace_financial_amounts import token
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class ReplaceFinancialAmount(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["en"]
+
+    def __init__(
+        self, seed: int = 0, max_outputs: int = 1, preprocessing: bool = True
+    ) -> None:
+        super().__init__(seed=seed, max_outputs=max_outputs)
+        self.financial_amount_transformation = (
+            token.FinancialAmountTransformation(
+                seed, max_outputs, preprocessing
+            )
+        )
+
+    def generate(self, sentence: str) -> List[str]:
+        # get the generator state to keep it the same for all replacement within the text to ensure similar output format
```

Comment:
i think this sentence needs to be re-worked!

Reply:
sentence modified